pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -115,6 +115,10 @@ def __init__(
         torch_inf = torch.tensor(np.Inf)
         self.best_score = torch_inf if self.monitor_op == torch.lt else -torch_inf
 
+    @property
+    def state_identifier(self) -> str:
+        return self._generate_state_identifier(monitor=self.monitor)
```

Comment:
nit: do we want a setter for state_identifier too?

Reply:
Good question. Do we want users to define their state_id directly in the constructor, like

```
def __init__(self, ...):
    ...
    self.state_id = ...
```
?

Reply:
Could we do something like this ?

```py
from pytorch_lightning.utilities.parsin import save_hyperparameters

def __init__(self, ...):
     save_hyperparameters(self)

def _ state_id(self):
       if hasattr(self, "hparams"):
            return hash(self.hparams):
       return self.state_id
```

Reply:
@tchaton Hashing all parameters is unfortunately not so practical for users. We discussed it a little bit here:  #6467
The issue is that there are certain input parameters in the callbacks (e.g. the ModelCheckpoint) that users may wish to change when resuming from a checkpoint, but if the init arguments are captured by a hash then we won't find that state in the checkpoint if users change the inputs. 

state_id is supposed to take care of that by defining the unique possible instantiations for a callback that we will be able to load back the state.